src/components/UniversalAmount.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+    <span class="amount" :class="{ approx: showApprox && isApprox }">
+        {{ formattedAmount }}
+        <span class="currency" :class="currency">{{currency}}</span>
+    </span>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import bigInt from 'big-integer';
+
+@Component
+export default class UniversalAmount extends Vue {
+    /**
+     * Amount in smallest unit
+     */
+    @Prop({type: [Number, Object]}) public amount!: number | bigInt.BigInteger;
+    /**
+     * Actual decimal count of the currency being displayed
+     */
+    @Prop({type: Number, default: 5}) public decimals!: number;
+    @Prop({type: Number, default: 2}) public minDecimals!: number;
+    @Prop({type: Number, default: 5}) public maxDecimals!: number;
+    @Prop({type: Boolean, default: false}) public showApprox!: boolean;
+    @Prop({type: String, default: 'nim'}) public currency;
+
+    private isApprox: boolean = false;
+
+    private get formattedAmount() {
+        let result: string;
+
+        if (typeof this.amount === 'number') {
```

Comment:
The calculation on numbers potentially leads to precision loss.
I'd suggest to convert the number to a `bigInt` and use the `bigInt` calculation for all inputs.
Note that `bigInt` also supports native `bigint`s as input althought that's not reflected in the types yet.
Thus, would need a `@ts-ignore` here if we want to support native `bigint` (which would be good in my opinion).

Reply:
The `big-integer` project maintainer already merged and published my type update PR which added native `bigint`s as valid input types for `bigInt`.
So, we can use that now.